sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/view/MeterSelector.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.metrics.view;
+
+import com.google.auto.value.AutoValue;
+import java.util.Objects;
+import java.util.regex.Pattern;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * Provides means for selecting one or more Meters. Used for selecting instruments when constructing
+ * views.
+ */
+@AutoValue
+@Immutable
+public abstract class MeterSelector {
+  private static final Pattern MATCH_ALL = Pattern.compile(".*");
```

Comment:
Ideally Java did it for us but it doesn't... Performance for simple selectors would go up a lot if we define a `Matcher` and don't use regex for matchAll or exact matches.

Maybe inspiration

https://github.com/open-telemetry/opentelemetry-java-contrib/blob/main/aws-xray/src/main/java/io/opentelemetry/contrib/awsxray/SamplingRuleApplier.java#L329

Reply:
Agreed.  I might use that directly against Predicate<String> for now.  It's almost exactly what you have but leveraging a more powerful `Predicate` class.  LMKWYT.